--- /usr/local/directadmin/scripts/letsencrypt.sh	2017-03-28 19:07:21.000000000 +1300
+++ /usr/local/directadmin/scripts/letsencrypt_poralix.sh	2017-04-11 16:28:31.278336745 +1200
@@ -494,10 +494,26 @@
 	
 	send_signed_request "normal" "${CHALLENGE_URI}" "{\"resource\": \"challenge\", \"keyAuthorization\": \"${KEYAUTH}\"}"
 
-	if [ ${HTTP_STATUS} -ne 202 ] ; then
-		echo "Challenge error: ${RESPONSE}. Exiting..."
-		exit 1
-	fi
+	while [ ${HTTP_STATUS} -ne 202 ] ; do
+		echo "Challenge error: ${RESPONSE}."
+		echo "";
+		echo -n "Do you want to retry (yes/no): "
+		while read yesno;
+		do
+			case "${yesno}" in
+				yes|y)
+					send_signed_request "normal" "${CHALLENGE_URI}" "{\"resource\": \"challenge\", \"keyAuthorization\": \"${KEYAUTH}\"}"
+					break;
+				;;
+				no|n)
+					exit 1;
+				;;
+				*)
+					echo -n "Do you want to retry (yes/no): "
+				;;
+			esac;
+		done;
+	done
 
 	echo "Waiting for domain verification..."
 	while [ "${CHALLENGE_STATUS}" = "pending" ]; do
@@ -515,6 +531,13 @@
 		echo "Challenge is ${CHALLENGE_STATUS}. Details: ${CHALLENGE_DETAIL}. Exiting..."
 		exit 1
 	fi
+	
+	echo -n "Sleep some ";
+	for n in `seq 1 5`; do
+		sleep 1;
+		echo -n '.';
+	done;
+	echo '';
 }
 done
 
